Restore the db `_fullPath` upon close()

keckler opened the pull request:
## Description

When a db open in write-mode is closed, it is moved out of the `FAST_PATH` back into the working directory. However, it's internal metadata `database._fullPath` was not updated to reflect this new path. 

This PR just updates the `_fullPath` attribute to be consistent.

This helps with keeping track of the DB during a run.

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

<!-- Check the code quality -->

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

<!-- Check the project-level cruft -->

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

john-science commented:
I feel like this should break a lot of things.  Don't many people downstream of ARMI expect the output DB to be in a certain place after a run?

Have you tested this with our major downstream applications, to make sure it doesn't break any of the popular simulation workflows?

I ask because (a) that would be an important thing to know, and (b) this PR seems more like a "nice to have" than a "need".

keckler commented:
I'm not moving where the output DB is. It has always been moved out of the FAST_PATH and back to the working directory.

This did not change.

I simply updated the internal tracker to reflect this change. Before, the `database._fullPath` attribute was left in a state saying that the file was in the `FAST_PATH` even after it had been moved out. That was a bug. So I fixed that.

Further, this doesn't even happen until the database is closed, which in a normal run only happens in `interactEOL()` of the `database` interface, after which no further analysis is done during a typical run.

So I do not expect this to impact anybody's run negatively. The only reason I even found this bug was because I added a `UserPlugin` to my run which takes place _after_ the `database` interface at EOL. (It does not break my run when the `UserPlugin` is disabled either)

Does that address your concerns? Or if no, maybe we can talk offline.

john-science commented:
@keckler  Oh, sorry, that was a misread.  I thought you _added_ the `shutil.move()` line, and was immediately SO concerned I stopped reading and didn't notice that all you did was _change_ that line to include the return value.

Carry on.
